test/e2e/e2e_test.go

```diff
@@ -589,6 +589,96 @@ func pollForResource(t *testing.T, resourceID string, timeout time.Duration, acc
 	return obj, err
 }
 
+func checkClientPodRcvdUpdatedServerCert(t *testing.T, client *kubernetes.Clientset, service *v1.Service, testNS, updatedServerCert string) {
+	timeout := 5 * time.Minute
+	metricsURL := fmt.Sprintf("%s.%s.svc:%d", service.Name, service.Namespace, service.Spec.Ports[0].Port)
+	err := wait.PollImmediate(10*time.Second, timeout, func() (bool, error) {
+		podName := "client-pod-" + randSeq(5)
+		_, err := client.CoreV1().Pods(testNS).Create(&v1.Pod{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: testNS,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "cert-checker",
+						Image:   "nicolaka/netshoot:latest",
+						Command: []string{"/bin/bash"},
+						Args:    []string{"-c", fmt.Sprintf("openssl s_client -showcerts -connect %s < /dev/null 2>/dev/null | openssl x509", metricsURL)},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyOnFailure,
+			},
+		})
+		defer deletePod(client, podName, testNS)
```

Comment:
As per a previous comment, it's still necessary to check error conditions when calling something with `defer`. As written here, any error will be ignored. At the least, it should be logged:

```golang
defer func() {
  err := deletePod(...)
  if err != nil {
  ...
  }
}
```

Another option - have the deferred method accept `*testing.T` and log/fail/whatever directly.

Reply:
This is my stupidity! corrected it now!